src/index.ts

```diff
@@ -98,18 +116,39 @@ import * as t from 'io-ts'
 
 `
 
+const newtypePrelude = `
+interface Newtype<URI, A> {
+  _URI: URI
+  _A: A
+}
+interface Iso<S, A> {
+  unwrap: (s: S) => A
+  wrap: (a: A) => S
+}
+const unsafeCoerce = <A, B>(a: A): B => a as any
+type Carrier<N extends Newtype<any, any>> = N['_A']
+type AnyNewtype = Newtype<any, any>
+const fromNewtype: <N extends AnyNewtype>(type: t.Type<t.mixed, Carrier<N>>) => t.Type<t.mixed, N> =
+  type => type as any
+const iso = <S extends AnyNewtype>(): Iso<S, Carrier<S>> =>
+  ({ wrap: unsafeCoerce, unwrap: unsafeCoerce })
+
+`
+
 export function getModels(models: Array<Model>, options: GetModelsOptions, prelude: string = getModelsPrelude): string {
-  const newtypes: { [key: string]: true } = {}
-  options.newtypes.forEach(k => {
-    newtypes[k] = true
-  })
-  const declarations = getDeclarations(models, options.isReadonly, options.optionalType || gen.undefinedType, newtypes)
-  const sortedDeclarations = gen.sort(sortBy(declarations, ({ name }) => name))
+  const declarations = getDeclarations(models, options.isReadonly, options.optionalType || gen.undefinedType)
+  const newtypeDeclarations: NewtypeRawDeclaration[] = declarations.filter((d): d is NewtypeRawDeclaration => d.kind === 'newtype')
+  const typeDeclarations: gen.TypeDeclaration[] = declarations.filter((d): d is gen.TypeDeclaration => d.kind !== 'newtype')
```

Comment:
here's we split proper io-ts-gen declarations and "raw" ones. This "split" is the reason why we can't sort newtypes declarations at the moment